### DISCUSSION TITLE
Integrate Fief into existing FastAPI app or run Fief as a separate app?

### DISCUSSION THREAD

**Discussion by kielerrr** [fief-dev/fief#325] [Q&A] [answered]
I have an existing Vue3/FastAPI app. Is the strategy generally to combine the existing FastAPI app endpoints and Fief endpoints together into a single FastAPI backend instance? 
Or is the general strategy to have the existing app and Fief running as separate FastAPI apps?
My existing app has 20+ endpoints already and integrating the Fief directly into the app will another 20 endpoints. 

Edit
I found the answer to my own question. It appears the answer is to run Fief as a standalone app and reference it in my existing app like
```
fief = FiefAsync(
    "https://my-fief-instance-url:8000",
    "YOUR_CLIENT_ID",
    "YOUR_CLIENT_SECRET",
)
```

**Comment by fief-bailiff** [comment]
Hail, @kielerrr 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment, marked as answer]
Hi @kielerrr 👋 

That's actually a very interesting question 😄

As you guessed, Fief is designed to be run as a standalone app (would I dare to say like a "microservice") running alongside your own apps.

That said, now we're currently re-architecturing the app to drop multiple workspaces support (#323), and since Fief is a pure FastAPI/ASGI app, I also think it will be possible to *mount* it directly into another FastAPI/ASGI app so everything runs inside the same process. That's not a reality right now, but I think I'll document this use-case when it becomes possible.
